### PR TITLE
feat: Add getColorByInternalName utility function

### PR DESCRIPTION
**Pull request by engfragui**
### Overview 

While working on upgrading `Todoist-Outlook` to Todoist API v2 by leveraging `todoist-api-typescript` v2 (https://github.com/Doist/Todoist-Outlook/pull/609), I've noticed that we don't currently have a way to retrieve a `Color` object (https://github.com/Doist/todoist-api-typescript/blob/main/src/types/entities.ts#L149-L153) via its "internal name", which is what is saved in the `Project.color` field.

As an example, there's no way to map "berry_red" to a `Color`.

This PR adds a utility function that does that (+ some simple tests).

### Reference 

Blocks https://github.com/Doist/Todoist-Outlook/pull/609

### Alternatives considered

We (briefly?) considered restructuring the `Color` type so that its `name` would map to the "internal name" (i.e. "berry_red") and `displayName` to its... errr... display name (i.e. "Berry Red"). However, this would be a breaking change and would require bumping `todoist-api-typescript` to v3.

### CODE REVIEW

File: src/types/entities.ts [hunk 1]
```diff
@@ -149,6 +149,7 @@ export type User = Static<typeof User>
 export type Color = {
     id: number
     name: string
+    internalName: string
     value: string
 }
```
[the remark this answers is not in the document]
Comment (scottlovegrove) [reply src/types/entities.ts:154]: I like this approach.

File: src/utils/colors.ts [hunk 2]
```diff
@@ -44,12 +89,19 @@ export const colors = [
...
+    return color ?? defaultColor
 }
 
 export function getColorByName(colorName: string): Color {
     const color = colors.find((color) => color.name === colorName)
-    return color ?? charcoal
+    return color ?? defaultColor
+}
+
+export function getColorByInternalName(colorInternalName: string): Color {
```
[the remark this answers is not in the document]
Comment (scottlovegrove) [reply src/utils/colors.ts:104]: Either this, or `getColorByApiName`, I'm not in favour of having "Internal" in an *external* API name.

**Review by scottlovegrove — APPROVED** [review]
I agree with @pawelgrimm's suggestsions.

File: src/utils/colors.ts [hunk 3]
```diff
@@ -44,12 +89,19 @@ export const colors = [
...
     const color = colors.find((color) => color.id === colorId)
-    return color ?? charcoal
+    return color ?? defaultColor
 }
 
 export function getColorByName(colorName: string): Color {
     const color = colors.find((color) => color.name === colorName)
-    return color ?? charcoal
+    return color ?? defaultColor
+}
```
[the remark this answers is not in the document]
Comment (scottlovegrove) [reply src/utils/colors.ts:102]: Essentially these could be considered being marked as deprecated. They were required for the old API as we returned an id, and the `getColorByName` was only added in v2 of the SDK based on an incorrect assumption by me that the Name was the pretty name, and not the snake_case name.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@doist/todoist-api-typescript",
-    "version": "2.0.7",
+    "version": "2.0.8",
```
[the remark this answers is not in the document]
Comment (engfragui) [reply package.json]: I think before marking the different `Color`s as `const` we didn't have any non-backward compatible changes, but I might be wrong... 🤔

**Comment by engfragui** [comment]
Just to clarify, I integrated all your great suggestions in this PR and will now use these changes in my `Todoist-Outlook` update. Thank you for spending time looking at this, much appreciated.

**Pull request merged**